sycl/doc/extensions/experimental/sycl_ext_oneapi_graph.asciidoc

```diff
@@ -1011,6 +1016,146 @@ If a node containing a dynamic parameter is updated through the whole graph
 update API, then any previous updates to the dynamic parameter will be reflected
 in the new graph.
 
+==== Graph-Owned Memory Allocations [[graph-memory-allocations]]
+
+It can be desirable for a graph to own and manage memory allocations for memory
+associated with commands in the graph. This is made possible by using the
+`async_malloc` and `async_free` functions from the
+link:../experimental/sycl_ext_oneapi_async_memory_alloc.asciidoc[sycl_ext_oneapi_async_memory_alloc
+] extension. These commands can be added to a graph either via queue recording
+or explicit graph creation, which will create allocations which are owned and
+managed by that specific `command_graph`.
+
+Graphs use vitual memory to manage these allocations. To use this feature in a
+graph the device associated with the graph must support the
+`ext_oneapi_virtual_mem` aspect. Attempting to use this feature with devices
+that do not support this aspect will result in synchronous errors being thrown.
+
+Pointers returned from allocation nodes can be used in other graph nodes in the
+same way as regular USM pointers.
+
+===== Supported Features [[allocation-supported-features]]
+
+Currently only device allocations are supported in graphs. Attempting to
+add allocations of any other type to a graph will result in synchronous errors
+being thrown.
+
+===== Restrictions
+
+The following restrictions apply to any graph using these allocations:
+
+* The graph cannot be finalized more than once.
+* The graph cannot be used as a sub-graph in another graph.
+* Graph memory allocation nodes cannot be updated, and graphs containing these
```

Comment:
A bit off topic:

I undestand why this limitation to update exists. But I wonder if it will limit the practical usecases of this functionality. Is it common for applications to be happy to just reuse the same graph without ever needing to update it?

Reply:
As a consequence of not being able to update allocation nodes it does limit the ability to use whole-graph update specifically, but you can still use explicit graph update on other nodes in a graph which don't have this limitation.

Whole-graph update is problematic I think because the use of the function creates this promise/contract that all of the nodes in the executable graph will be updated to match the state of the source modifiable graph. We could skip allocation nodes during the update but that creates a problem since that modifiable graph will have different virtual addresses that are used in the other nodes in the graph. 

It also breaks that promise/contract which is somewhat unintuitive, since the state of the executable graph will not actually match the source graph.